### PR TITLE
Model related docs refine

### PR DESCRIPTION
**Pull request by CoderKevinZhang**
# Description

Refine all Model related docs with descriptions and example usage

### CODE REVIEW

**Review by diondrapeck — CHANGES_REQUESTED** [review]

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/model_batch_deployment_settings.py [hunk 1]
```diff
@@ -39,6 +39,15 @@ class ModelBatchDeploymentSettings:
     :type error_threshold: int, optional
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_deployment/model_batch_deployment_settings.py:39]: Fixed

File: sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/_package/model_package.py [hunk 2]
```diff
@@ -192,6 +201,15 @@ class ModelPackage(Resource, PackageRequest):
     :type model_configuration: Optional[~azure.ai.ml.entities.ModelConfiguration]
     :param tags: The tags of the model package.
     :type tags: Optiona[dict[str, str]]
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/entities/_assets/_artifacts/_package/model_package.py]: Fixed

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py [hunk 3]
```diff
@@ -311,6 +337,14 @@ def download(self, name: str, version: str, download_path: Union[PathLike, str]
         :param Union[PathLike, str] download_path: Local path as download destination,
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py]: Fixed

File: sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py [hunk 4]
```diff
@@ -424,11 +474,22 @@ def list(
...
-        :keyword list_view_type: View type for including/excluding (for example) archived models. Default: ACTIVE_ONLY.
-        :type list_view_type: Optional[ListViewType]
+        :type name: Optional[str], optional
+        :param stage: The Model stage
+        :type stage: Optional[str], optional
+        :keyword list_view_type: View type for including/excluding (for example) archived models. Defaults to
+             :attr:`ListViewType.ACTIVE_ONLY`.
+        :type list_view_type: ListViewType, optional
         :return: An iterator like instance of Model objects
         :rtype: ~azure.core.paging.ItemPaged[Model]
```
[the remark this answers is not in the document]
Comment (CoderKevinZhang) [reply sdk/ml/azure-ai-ml/azure/ai/ml/operations/_model_operations.py]: Fixed

**Review by diondrapeck — APPROVED** [review]

**Comment by CoderKevinZhang** [comment]
> @CoderKevinZhang please read the following Contributor License Agreement(CLA). If you agree with the CLA, please reply with the following information.
> 
> ```
> @microsoft-github-policy-service agree [company="{your company}"]
> ```
> 
> > Options:
> > 
> > * (default - no company specified) I have sole ownership of intellectual property rights to my Submissions and I am not making Submissions in the course of work for my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > 
> > * (when company given) I am making Submissions in the course of work for my employer (or my employer has intellectual property rights in my Submissions by contract or applicable law). I have permission from my employer to make Submissions and enter into this Agreement on behalf of my employer. By signing below, the defined term “You” includes me and my employer.
> > 
> > ```
> > @microsoft-github-policy-service agree company="Microsoft"
> > ```
> 
> Contributor License Agreement

**Pull request merged**